daemon/src/db.rs

```diff
@@ -431,6 +455,37 @@ impl Connection {
 
         Ok(ids)
     }
+
+    async fn failed_cfd_ids_according_to_events(&self) -> Result<Vec<OrderId>> {
+        let mut conn = self.inner.acquire().await?;
+
+        let ids = sqlx::query!(
+            r#"
+            select
+                id as cfd_id,
+                uuid as "uuid: model::OrderId"
+            from
+                cfds
+            where exists (
+                select id from EVENTS as events
+                where events.cfd_id = cfds.id and
+                (
+                    events.name = $1 or
+                    events.name = $2
+                )
+            )
+            "#,
+            model::EventKind::OFFER_REJECTED,
+            model::EventKind::CONTRACT_SETUP_FAILED,
+        )
+        .fetch_all(&mut *conn)
+        .await?
+        .into_iter()
+        .map(|r| r.uuid)
+        .collect();
```

Comment:
Here we could omit the `into_iter` part of we would use `fetch`.

Reply:
I don't seem to be able to return the stream because `conn` is borrowed and dropped as soon as we return :(

```rust
// error[E0515]: cannot return value referencing local variable `conn`
//    --> daemon/src/db.rs:391:9
//     |
// 388 |         .fetch(&mut *conn)
//     |                      ---- `conn` is borrowed here
// ...
// 391 |         Ok(ids)
//     |         ^^^^^^^ returns a value referencing data owned by the current function

async fn closed_cfd_ids_according_to_the_blockchain(
    &self,
) -> Result<impl Stream<Item = Result<OrderId, sqlx::Error>>> {
    let mut conn = self.inner.acquire().await?;

    let ids = sqlx::query!(
        r#"
        select
            id as cfd_id,
            uuid as "uuid: model::OrderId"
        from
            cfds
        where exists (
            select id from EVENTS as events
            where events.cfd_id = cfds.id and
            (
                events.name = $1 or
                events.name = $2 or
                events.name= $3
            )
        )
        "#,
        EventKind::COLLABORATIVE_SETTLEMENT_CONFIRMED,
        EventKind::CET_CONFIRMED,
        EventKind::REFUND_CONFIRMED,
    )
    .fetch(&mut *conn)
    .map(|res| res.map(|r| r.uuid));

    Ok(ids)
}

```